### DISCUSSION TITLE
Turn laser on immediately when M3 command is issued interactuively

### DISCUSSION THREAD

**Discussion by frankvdh** [arkypita/LaserGRBL#975] [Q&A]
When I issue the M3 S1 command interactively, it doesn't actually turn the laser on until a move command is executed. That confused me for a long time, thinking my laser wasn't working.

**Comment by gmmanonymus111** [comment]
That is the default behaviour of GRBL when lasermode (`$31`) is set to 1, and LaserGRBL can't override it. Your best bet is make a button which sets `$31` to 0 while you set focus, and when you finished, sets it back to 1. Like this TwoStateButton:
ON:
```
$32=0
M3 S1
```
OFF:
```
M5 S0
$32=1
```
